### PR TITLE
add support for private scoped packages

### LINKED ISSUE #12: private scoped packages always return 404
If there is a private scoped package you have access to, the registry will return the info instead of a 404.

However, it seems that if auth data is not included in the request, the registry doesn't know who you are and will always return a 404 for private packages.

### PR DESCRIPTION
**Pull request by sindresorhus**
fixes #12 

// @dylang @homerjam 

I don't have any way to test this, though, so would be nice if you could confirm. You can edit the scoped package test to one of your own private scope packages.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
 var registryUrl = require('registry-url');
+var rc = require('rc');
 
 module.exports = function (name, version) {
-	var url = registryUrl(name.split('/')[0]) +
+	var scope = name.split('/')[0];
+	var url = registryUrl(scope) +
 		encodeURIComponent(name).replace(/^%40/, '@');
+	var npmrc = rc('npm');
+	var token = scope ? npmrc[scope + ':_authToken'] : npmrc['//registry.npmjs.org/:_authToken'];
```
Comment (dylang) [inline index.js]: FWIW, I don't have `_authToken` in my `.npmrc`, i have `_authCrypt`.

Comment (dylang) [inline index.js]: I'm not sure how npm is working for me. Will dig around.
http://blog.npmjs.org/post/118393368555/deploying-with-npm-private-modules

Comment (dylang) [inline index.js]: Ah, I have the `//registry.npmjs.org/:_authToken`, I'm just tried.

Comment (dylang) [inline index.js]: If I change the code to 

```
var token = npmrc['//registry.npmjs.org/:_authToken'];
```

It works.

Comment (sindresorhus) [inline index.js]: How about now?

The `npmrc[scope + ':_authToken']` there for people that have scoped privately hosted modules.

**Comment by sindresorhus** [comment]
@dylang How about now?

The `npmrc[scope + ':_authToken']` there for people that have scoped privately hosted packages.

**Comment by homerjam** [comment]
:+1: This is working for me

**Comment by dylang** [comment]
:+1: worked for me as well, great work @sindresorhus!

**Pull request merged**